R/filter.R

```
@@ -0,0 +1,218 @@
+#' used to evaluate the expression), and are defined in terms of the variables
+#' in .data. If multiple expressions are included, they are combined with the &
+#' operator. Only rows for which all conditions evaluate to TRUE are kept.
+#'
+#' @export
+#'
+#' @return A summarised_result object with only the rows that fulfill the
+#' required specified additional.
+#'
+#' @examples
+#' library(dplyr)
+#' library(omopgenerics)
+#'
+#' x <- tibble(
+#'   "result_id" = 1L,
+#'   "cdm_name" = "eunomia",
+#'   "group_name" = "cohort_name",
+#'   "group_level" = c("cohort1", "cohort2", "cohort3"),
+#'   "strata_name" = "sex",
+#'   "strata_level" = "Female",
+#'   "variable_name" = "number subjects",
+#'   "variable_level" = NA_character_,
+#'   "estimate_name" = "count",
+#'   "estimate_type" = "integer",
+#'   "estimate_value" = c("100", "44", "14"),
+#'   "additional_name" = c("year", "time_step", "year &&& time_step"),
+#'   "additional_level" = c("2010", "4", "2015 &&& 5")
+#' ) |>
+#'   newSummarisedResult()
+#'
+#' x |>
+#'   filterAdditional(year == "2010")
+#'
+filterAdditional <- function(result, ...) {
+  filterNameLevel(result, "additional", ...)
+}
+
+filterNameLevel <- function(result, prefix, ..., call = parent.frame()) {
+  # initial checks
+  cols <- paste0(prefix, c("_name", "_level"))
+  omopgenerics::assertTable(result, columns = cols, call = call)
+
+  # splitNameLevel
+  labs <- result |>
+    dplyr::select(dplyr::all_of(cols)) |>
+    dplyr::distinct() |>
+    splitNameLevel(name = cols[1], level = cols[2], keep = TRUE)
+
+  # filter
+  tryCatch(
+    expr = {
+      result |>
+        dplyr::inner_join(
+          labs |>
+            dplyr::filter(...) |>
+            dplyr::select(dplyr::all_of(cols)),
+          by = cols
+        )
+    },
+    error = function(e) {
+      cli::cli_warn(c(
+        "!" = "Variable filtering does not exist, returning empty result: ",
```

I would throw an error here as well